test/org/sagebionetworks/bridge/models/schedules/CronActivitySchedulerTest.java

```diff
@@ -133,6 +135,54 @@ public void recurringCronScheduleAgainstCalculatedEventWorks() {
         assertEquals(ScheduledActivityStatus.AVAILABLE, scheduledActivities.get(0).getStatus());
         DateTimeUtils.setCurrentMillisSystem();
     }
+    @Test
+    // All the dates in this test are derived from BRIDGE-1211.
+    public void recurringCronScheduleWithStartEndTimeWindowWorks() {
+        DateTime now = DateTime.parse("2016-03-15T17:13:13.044Z");
+        DateTimeUtils.setCurrentMillisFixed(now.getMillis());
+        
+        Schedule schedule = createScheduleWith(RECURRING);
+        schedule.setCronTrigger("0 0 0 ? * MON *");
+        schedule.setEventId("two_weeks_before_enrollment");
+        schedule.setStartsOn(DateTime.parse("2016-03-14T00:00:00.000Z"));
+        schedule.setEndsOn(DateTime.parse("2016-03-20T23:59:00.000Z"));
+        schedule.setExpires("P7D");
+        
+        events.clear();
+        events.put("enrollment", DateTime.parse("2016-03-02T00:04:37.000Z"));
+        events.put("two_weeks_before_enrollment", events.get("enrollment").minusWeeks(2));
+        
+        // We'd expect, based on that schedule, to have a task:
+        scheduledActivities = schedule.getScheduler().getScheduledActivities(plan, getContext(now.plusDays(4)));
+        // 1 or 2 activities depending on the time of the week (4 day window creates some overlap)
+        assertTrue(scheduledActivities.size() == 1 || scheduledActivities.size() == 2);
```

Comment:
Isn't there exactly 1 Monday between Mar 14 and Mar 20? And don't we know the exact schedule start dates and end dates? Shouldn't we be able to determinsitically say how many events are in the schedule?